### DISCUSSION TITLE
UX: Maybe subcommands should be the 1st argument instead of options

### DISCUSSION THREAD

**Discussion by noncombatant** [FiloSottile/age#243] [UX reports]
This is an arguable style matter. But I find the 'modern' style of `go` and `git` a bit easier to use. E.g.:

`age decrypt options...`

`age encrypt options...`

Then you'd get rid of `-d`, `--decrypt`, and `-r` and `-p` would no longer implicitly mean "encrypt". It'd be explicit.

**Comment by noncombatant** [comment]
I should also add: This allows the possibility of combining `age` and `age-keygen`, by creating an `age keygen` subcommand. Then there'd only be 1 binary, and possibly more code sharing.

**Comment by ovalseven8** [comment]
See #22

**Comment by FiloSottile** [comment]
There is now an optional `--encrypt` for being explicit in scripts, but probably too late to switch to subcommands.
